test-harness/src/test/groovy/com/netflix/conductor/test/resiliency/QueueResiliencySpec.groovy

```diff
@@ -0,0 +1,557 @@
+        0 * queueDAO._
+        with(workflowResource.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+    }
+
+    def "Verify reset callbacks fails when QueueDAO is unavailable"() {
+        when: "Start a simple workflow"
+        def workflowInstanceId = workflowResource.startWorkflow(new StartWorkflowRequest()
+                .withName(SIMPLE_TWO_TASK_WORKFLOW)
+                .withVersion(1))
+
+        then: "Verify workflow is started"
+        with(workflowResource.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        when: "Task is updated with callBackAfterSeconds"
+        def workflow = workflowResource.getExecutionStatus(workflowInstanceId, true)
+        def task = workflow.getTasks().get(0)
+        def taskResult = new TaskResult(task)
+        taskResult.setCallbackAfterSeconds(120)
+        taskResource.updateTask(taskResult)
+
+        and: "and then reset callbacks when QueueDAO is unavailable"
+        workflowResource.resetWorkflow(workflowInstanceId)
+
+        then: "Verify an exception is thrown"
+        1 * queueDAO.resetOffsetTime(*_) >> { throw new ApplicationException(ApplicationException.Code.BACKEND_ERROR, "Queue resetOffsetTime failed from Spy") }
+        thrown(ApplicationException)
+    }
+
+    def "Verify search is not impacted by QueueDAO"() {
+        when: "We perform a search"
+        workflowResource.search(0, 1, "", "", "")
+
+        then: "Verify it doesn't involve QueueDAO"
+        0 * queueDAO._
+    }
+
+    def "Verify search workflows by tasks is not impacted by QueueDAO"() {
+        when: "We perform a search"
+        workflowResource.searchWorkflowsByTasks(0, 1, "", "", "")
+
+        then: "Verify it doesn't involve QueueDAO"
+        0 * queueDAO._
+    }
+
+    def "Verify get external storage location is not impacted by QueueDAO"() {
+        when:
+        workflowResource.getExternalStorageLocation("", "", "")
+
+        then: "Verify it doesn't involve QueueDAO"
+        0 * queueDAO._
+    }
+
+
+    /// Task Resource endpoints
+
+    def "Verify polls return with no result when QueueDAO is unavailable"() {
+        when: "Some task 'integration_task_1' is polled"
+        def pollResult = taskResource.poll("integration_task_1", "test", "")
+
+        then:
+        1 * queueDAO.pop(*_) >> { throw new IllegalStateException("Queue pop failed from Spy") }
+        0 * queueDAO._
+        notThrown(Exception)
+        pollResult == null
+    }
+
+    def "Verify updateTask succeeds when QueueDAO is unavailable"() {
```

Comment:
Do you want to add another test case for updateTask with IN_PROGRESS state? 
This case only covers the updateTask with COMPLETED state, which does not directly use the queue, but the former uses queue to postpone.